### DISCUSSION TITLE
I'm not able to start certain apps in win-kex

### DISCUSSION THREAD

**Discussion by martin15560** [microsoft/WSL#6117] [General]
# Environment

```none
Windows build number: Windows 10 version 2004 build 19041
Your Distribution version: Kali 2020.3
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.128-microsoft-standard (WSL 2)
```

#  Expected behavior

I would like to be able to start "Window Manager" and "Disks" in Accessories and "Wireshark" app. There are probably other apps which can't be run also. 

# Actual behavior

When I click on the icons in Win-Kex environment in menu panel nothing happens.

**Comment by LRitzdorf** [comment]
I noticed the same issue with Wireshark on my WSL2 install. The "start menu" entry for Wireshark runs the command `pkexec wireshark`, so I tested pkexec in the terminal, with the following result:
```
$ pkexec wireshark
Error getting authority: Error initializing authority: Could not connect: No such file or directory
```

Also, when I raised the issue in #6178, the response mentioned that dbus and/or polkit could be the cause. I would think this shouldn't be the case, since I ran `sudo apt install kali-linux-default` (which installs all the standard Kali tools), but I don't know, since I have no experience with either dbus or polkit.

**Comment by sdancer75** [comment]
Same problem here

**Reply by LRitzdorf** [reply]
Initial question: do you see a `dbus.service` systemd unit on your Kali install? If so, check on its status, and make sure it's started successfully. If not, install `dbus` and try again. Let us know if this helps!

**Reply by sdancer75** [reply]
Hi,

dbus was not running, but I did it manually. Nothing changed. WireShark didn't start.

**Comment by moritzbeck13** [comment]
Will paste this here aswell:

On Kali, running Wireshark from the application menu is linked to sh -c "$(id -Gn | grep -q wireshark || echo pkexec) wireshark %f"
On WSL, the command behind the start menu entry is sh -c "$(id -Gn | grep -q wireshark || echo pkexec) wireshark

From running this command in PowerShell, I can see two things:

the quotes are not properly closed
PowerShell complains about the ||
After starting dbus, which isn't in systemd on WSL for some reason, and running sh -c "$(id -Gn | grep -q wireshark | echo pkexec) wireshark", PowerShell still complains, but Wireshark starts as expected.
I think WSL has some problems interpreting the %f and just cuts of the string altogether. Not sure about the rest.

Funnily enough; this was working for me with an existing install but recent version of Windows 11, but not on a fresh install.
